TrkHitReco/src/TNTClusterer.cc

```diff
@@ -6,29 +6,46 @@
 namespace mu2e
 {
   TNTClusterer::TNTClusterer(const Config& config) :
-    dhit_       (config.hitDistance()),
-    dseed_      (config.seedDistance()),
-    dd_         (config.clusterDiameter()),
-    dt_         (config.clusterTime()),
-    maxHitdt_   (config.maxHitTimeDiff()),
-    maxDistSum_ (config.maxSumDistance()),
-    maxNiter_   (config.maxCluIterations()),
-    useMedian_  (config.medianCentroid()),
-    comboInit_  (config.comboInit()),
-    bkgmask_    (config.bkgmsk()),
-    sigmask_    (config.sigmsk()),
-    testflag_   (config.testflag()),
-    diag_       (config.diag())
+    dhitDist_         (config.hitDistance()),
+    dseedDist_        (config.seedDistance()),
+    dhitChi2_         (config.hitChi2()),
+    dseedChi2_        (config.seedChi2()),
+    dd_               (config.clusterDiameter()),
+    dt_               (config.clusterTime()),
+    minClusterHits_   (config.minClusterHits()),
+    maxHitdt_         (config.maxHitTimeDiff()),
+    maxDistSum_       (config.maxSumDistance()),
+    maxNiter_         (config.maxCluIterations()),
+    useMedian_        (config.medianCentroid()),
+    comboInit_        (config.comboInit()),
+    bkgmask_          (config.bkgmsk()),
+    sigmask_          (config.sigmsk()),
+    testflag_         (config.testflag()),
+    diag_             (config.diag()),
+    distMethod_       (config.distMethod())
   {
-     float minerr (config.minHitError());
-     float maxdist(config.maxDistance());
-     float trms   (config.timeRMS());
-
-     tbin_     =1.0;
-     dd2_      = dd_*dd_;
-     maxwt_    = 1.0f/minerr;
-     md2_      = maxdist*maxdist;
-     trms2inv_ = 1.0f/trms/trms;
+    float minerr (config.minHitError());
+    float maxdist(config.maxDistance());
+    float trms   (config.timeRMS());
+
+    tbin_     =1.0;
+    dd2_      = dd_*dd_;
+    maxwt_    = 1.0f/minerr;
+    md2_      = maxdist*maxdist;
+    trms2inv_ = 1.0f/trms/trms;
+
+    switch(distMethod_) {
+      case TNTClusterer::useDistance:
+        dhit_ = dhitDist_;
```

Comment:
i think its better to reuse the same config functions for both dist and chi2 mode parameters, otherwise we must set values that are never used.  It will be clear from the fcl which one  you are configuring, so what unit they are in.  'Distance' is effectively an abstract quantity.

Reply:
Ok, will combine the two. I felt like it will be confusing when looking only at the config file.